### DISCUSSION TITLE
ModelSpec not working as expected after recent commits

### DISCUSSION THREAD

**Discussion by yashsinha-aero** [danny-avila/LibreChat#4213] [Troubleshooting] [answered]
### What happened?

I made no changes to my code except updating to recent commits, and my modelSpec with custom OpenAI endpoint has stopped working. My gptPlugins model spec, that uses the OPENAI_REVERSE_PROXY base, still works fine, but my custom openAI endpoint returns a modelspec mismatch, despite having selected the Model.

### Steps to Reproduce

1. Create ModelSpec with custom endpoint, with OpenAI models.
2. Enable enforce and prioritize.

### What browsers are you seeing the problem on?

Safari, Microsoft Edge, Mobile (iOS), Mobile (Android)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
Try testing it after this PR is merged. 
https://github.com/danny-avila/LibreChat/pull/4214

It makes more sense to apply the configured preset, regardless of what the client sent, instead of checking for a mismatch.

**Reply by yashsinha-aero** [reply]
@danny-avila I merged this PR to my repo. I dont get the modelmismatch error anymore, but the display name in the chat shows the actual OpenAI configured name. For example, if my model spec’s name is “My custom AI”, then even though i can select that name from the models list, it shows “GPT-4o” as the name of the AI during the chat.
The modelspec is set to a custom OpenAI compatible endpoint with “chatgpt-4o-latest”.

**Reply by danny-avila** [reply]
That is a separate issue, thanks

**Reply by yashsinha-aero** [reply]
alright, thanks for the quick response. I will create an issue regarding this.

**Comment by maxesse** [comment]
I had the same issue with Azure OpenAI, and I got it fixed by setting enforce to false. I'm not 100% sure what that option does, if all the interface parameters are turned off anyways? It's ok right to leave enforce to false then?

**Reply by danny-avila** [reply]
Try updating, and you won't have an issue with `enforce` anymore. Up until yesterday, this field would check for a mismatch of client parameters vs. what the modelSpec defines. I decided this is not the best way to go about this, and `enforce` should instead apply whatever was set in YAML config, instead, even if the client parameters all match.
